src/battle/area_flo/5B1A10.c

```diff
@@ -14,4 +14,48 @@
 
 #include "common/StartRumbleWithParams.inc.c"
 
-INCLUDE_ASM(s32, "battle/area_flo/5B1A10", func_80218808_5B1CB8);
+ApiStatus func_80218808_5B1CB8(Evt* script, s32 isInitialCall) {
+    Actor* actor;
+    s32 actorType;
+    s32 value;
+
+    Bytecode* readPos = script->ptrReadPos;
+    ActorID actorId = evt_get_variable(script, *readPos++);
+
+    if (actorId == -0x7F) {
+        actorId = script->owner1.enemyID;
+    }
+
+    actor = get_actor(actorId);
+    actorType = actor->actorType;
+
+    // These gotos are weird, but they produce the least weird looking match
+    // that I have found. Alternatives are an
+    //    evt_set_variable(script, *readPos, 0);
+    // in every possible else of every if, or a top level switch with breaks
+    // instead of gotos.
+    //
+    // Likewise, combining the conditions in any way will break the compiler's
+    // choice of the bnezl instruction.
+    if (actorType != 0x47) {
```

Comment:
This is almost certainly a switch over actorType

Reply:
That was my first thought as well. I'll play around with it some more and see if I can get a clean match with a switch.

Reply:
I got a janky match by doing the following:

```c
    switch (actorType) {
        case 0x4D: case 0x4E: case 0x4F: case 0x50:
        case 0x51: case 0x52: case 0x53: case 0x54:
        case 0x55: case 0x56: case 0x57: case 0x58:
        case 0x59: case 0x5A: case 0x5B: case 0x5C:
        case 0x5D: case 0x5E: case 0x5F: case 0x60:
        case 0x61: case 0x62: case 0x63: case 0x64:
            evt_set_variable(script, *args, -1);
            break;

        case 0x47:
            if (actor->varTable[8] == 1) {
                evt_set_variable(script, *args, -1);
                break;
            }
        default:
            evt_set_variable(script, *args, 0);
    }
```

Any thoughts? Seems a bit better.

Reply:
I think that's better, yeah! although it does take up a lot of space, could you put each case statement on a separate line?

Reply:
A better version of this is in the final commit.